runtime/floats.c

```diff
@@ -172,6 +167,83 @@ CAMLexport void caml_Store_double_array_field(value val, mlsize_t i, double dbl)
 }
 #endif /* ! FLAT_FLOAT_ARRAY */
 
+#ifdef HAS_VSNPRINTF_L
+static value caml_alloc_sprintf_l(locale_t loc, const char * format, ...)
+{
+  va_list args;
+  char buf[128];
+  int n;
+  value res;
+
+#ifndef _WIN32
```

Comment:
```suggestion
#if !defined(_WIN32) || defined(_UCRT)
```

> Beginning with the UCRT in Visual Studio 2015 and Windows 10, `vsnprintf` is no longer identical to `_vsnprintf`. The `vsnprintf` function conforms to the C99 standard; `_vsnprintf` is kept for backward compatibility with older code.

The file `vcruntime.h` defines `_UCRT`.

Maybe the legacy MSVCRT code can be inlined into the general case under the guard of `defined(_WIN32) && !defined(_UCRT)`? (I don't know if there's a macro for msvcrt).

Reply:
We're using `_vsnprintf_l`, though - there is no `vsnprintf_l` in UCRT?

Reply:
Whoopsie!